src/library_pthread.js

```diff
@@ -155,19 +155,25 @@ var LibraryPThread = {
       PThread.runningWorkers = [];
     },
     freeThreadData: function(pthread) {
-      if (!pthread) return;
-      if (pthread.threadInfoStruct) {
+      // When sanitizers are enabled, free is normally instrumented to call
+      // sanitizer code that checks some things about pthreads. We are not
+      // ready for such calls, as this can be called after the sanitizers
+      // are finalized. Instead, call free directly.
+      withBuiltinMalloc(function () {
```

Comment:
Actually I think we can can instead just replace the calls to `_free` below with calls to `_emscripten_builtin_free`.. just like elsewhere in the file we call `_emscripten_builtin_malloc`.

The `withBuiltinMalloc` is mostly useful for calling entire functions that need to operate in both builtin and non-builtin modes.

Reply:
Done with commit 3c41330121480c8f270b3486eddcdf6ec841bc9e.